rootfs/scheduler/__init__.py

```diff
@@ -993,6 +993,10 @@ def _wait_until_pods_are_ready(self, namespace, container, labels, desired):  #
             logger.info('timed out ({}s) waiting for pods to come up in namespace {}'.format(timeout, namespace))  # noqa
 
         logger.info("{} out of {} pods in namespace {} are in service".format(count, desired, namespace))  # noqa
+        if count != desired:
+            # raising to allow operations to rollback
+            raise KubeException('Not enough pods in namespace {} came into service. '
+                                '{} out of {}'.format(namespace, count, desired))
```

Comment:
@kmala do you have any suggestions what information we could bubble up? Note that this can be more than one pod that can fail as we do deploy batches now